packages/es-components/src/components/controls/buttons/Button.js

```diff
@@ -1,107 +1,162 @@
+    border-color: ${props => props.colors.hoverBorderColor};
+    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075),
+      0 0 0 0.2rem ${props => props.colors.focusBoxShadowColor};
+    outline: 0;
   }
 
-  &:focus,
-  &:focus-within {
-    box-shadow: 0 2px 4px 4px ${props => props.theme.colors.inputFocus};
+  &:active,
+  &.pressed {
+    color: ${props => props.colors.activeTextColor};
+    background-color: ${props => props.colors.activeBgColor};
+    border-color: ${props => props.colors.activeBorderColor};
+    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.25);
   }
 
-  &:active {
-    box-shadow: none;
+  &.pressed {
+    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.25);
   }
 
-  &&:active .es-button__display {
-    box-shadow: 0 0 0 0 transparent;
-    top: 4px;
+  &:active:focus {
+    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.25),
+      0 0 0 0.2rem ${props => props.colors.focusBoxShadowColor};
   }
 
-  &[disabled],
-  &[disabled] .es-button__display {
+  &[disabled] {
+    color: #ccc;
     cursor: not-allowed;
-    opacity: 0.65;
+    background-color: #e6e6e6;
+    border-color: #e6e6e6;
+
+    &:hover {
+      background-color: #e6e6e6;
+      border-color: #e6e6e6;
+      color: #ccc;
+    }
 
     > * {
       pointer-events: none;
     }
   }
-
-  &[disabled]:hover .es-button__display {
-    background-color: ${props => props.variant.bgColor};
-  }
 `;
 
 const Button = React.forwardRef(function Button(props, ref) {
   const { children, styleType, size, block, ...other } = props;
   const theme = useTheme();
   const buttonSize = theme.buttonStyles.button.size[size];
   const variant = theme.buttonStyles.button.variant[styleType];
+  const isInheritedStyle = styleType === 'inherited';
+
+  const focusBoxShadowColor = tinycolor.mix(
+    variant.bgColor,
+    theme.colors.black,
+    14
+  );
+  focusBoxShadowColor.setAlpha(0.5);
+
+  const inheritedButtonColors = {
+    bgColor: 'inherited',
+    textColor: 'inherited',
+    borderColor: 'inherited',
+    hoverBgColor: 'inherited',
+    focusBoxShadowColor: theme.colors.gray4,
+    activeBgColor: 'inherited',
+    activeTextColor: 'inherited',
+    activeBorderColor: 'inherited',
+    hoverTextColor: 'inherited'
+  };
+
+  const calculatedButtonColors = {
+    bgColor: variant.bgColor,
+    textColor: getTextColor(variant.bgColor),
+    borderColor: variant.bgColor,
+    hoverBgColor: darken(variant.bgColor, 7.5),
+    hoverBorderColor: darken(variant.bgColor, 9.9),
+    focusBoxShadowColor: focusBoxShadowColor.toRgbString()
+  };
+  calculatedButtonColors.activeBgColor = darken(
+    calculatedButtonColors.hoverBgColor,
+    2.5
+  );
+  calculatedButtonColors.activeTextColor = getTextColor(
+    calculatedButtonColors.activeBgColor
+  );
+  calculatedButtonColors.activeBorderColor = darken(
+    calculatedButtonColors.hoverBgColor,
+    5
+  );
+  calculatedButtonColors.hoverTextColor = getTextColor(
+    calculatedButtonColors.hoverBgColor
+  );
```

Comment:
This looks much better now! I might take it even one step further and put it all in a local function called something like `getCalculatedButtonColors`.

Reply:
Done